File: ForegroundService/app/src/main/java/com/example/foregroundservice/MyApplication.java
package com.example.foregroundservice;

import android.app.Application;
import android.app.NotificationChannel;
import android.app.NotificationManager;
import android.os.Build;
//thuc hanh foreground service
//viết code xử lý bên trong này
public class MyApplication extends Application {
//    khởi tạo 1 hằng số
    public static final String CHANNEL_ID = "channel_service_example";

    @Override
    public void onCreate() {
        super.onCreate();
//        khởi tạo channel id
        createChannelNotification();
    }

    private void createChannelNotification() {
        if(Build.VERSION.SDK_INT >= Build.VERSION_CODES.O){
            NotificationChannel channel = new NotificationChannel(CHANNEL_ID, "Channel Service Example",
                    NotificationManager.IMPORTANCE_DEFAULT);
            NotificationManager manager = getSystemService(NotificationManager.class);
            if(manager != null){
                manager.createNotificationChannel(channel);
            }
        }
    }
}

File: ForegroundService_KOTLIN/app/src/main/java/com/example/foregroundservice_kotlin/MyApplication.kt
package com.example.foregroundservice_kotlin

import android.app.Application
import android.app.NotificationChannel
import android.app.NotificationManager
import android.os.Build


//thuc hanh foreground service
//viết code xử lý bên trong này
class MyApplication : Application() {
    override fun onCreate() {
        super.onCreate()
        //        khởi tạo channel id
        createChannelNotification()
    }

    private fun createChannelNotification() {
        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.O) {
            val channel = NotificationChannel(
                CHANNEL_ID, "Channel Service Example",
                NotificationManager.IMPORTANCE_DEFAULT
            )
            val manager = getSystemService(
                NotificationManager::class.java
            )
            manager?.createNotificationChannel(channel)
        }
    }

    companion object {
        //    khởi tạo 1 hằng số
        const val CHANNEL_ID: String = "channel_service_example"
    }
}
